### ISSUE TITLE
API Permissions

### ISSUE THREAD

**Issue by bdwyertech** [wazuh/wazuh#116]
When trying to delete an agent via the API, I receive the following:

```
"Wazuh-Python Internal Error: [Errno 13] Permission denied: '/var/ossec/queue/rids/009'"
```

Looks like `wazuh-api` runs as the `ossec` user, however the `ossec` group does not have write permissions in the `rids` folder

Running CentOS 7.3 with SELinux enabled

```
[root@wazuh-1 rids]# ls -lah /var/ossec/queue/rids/ | less
total 84K
drwxr-x---.  2 ossecr ossec 32K May  1 14:16 .
drwxr-x---. 11 root   ossec 136 Apr 30 15:21 ..
-rw-r--r--.  1 ossecr ossec   7 May  1 14:10 001
-rw-r--r--.  1 ossecr ossec   7 May  1 11:26 002
......
-rw-r--r--.  1 ossecr ossec   0 May  1 11:37 996
-rw-r--r--.  1 ossecr ossec   0 May  1 11:37 997
-rw-r--r--.  1 ossecr ossec   0 May  1 11:37 998
-rw-r--r--.  1 ossecr ossec   0 May  1 11:37 999
```

**Comment by jlruizmlg** [comment]
Hi @bdwyertech after take a look with the developer team, you are right, Wazuh-API cannot remove this files because the parent folder has a incorrect permissions, your pull request is correct, in the next packages release this problem will be fixed.

Now if you change manually the permissions to 770 in this two folders, the problem should be fixed.

**Comment by bdwyertech** [comment]
@jlruizmlg I think it is a umask issue as well, the umask needs to be set to 0002 for the remoted function (whatever the ossecr user is for)

I found that even after fixing the permissions, new RID and agent-info files were being created as 0644. A umask of 0002 should cause the daemon to create them with 0660 permissions.

**Comment by bdwyertech** [comment]
I searched through the code and saw a few references to setting the umask, but nothing obviously in relation to the remoted daemon.

**Comment by bdwyertech** [comment]
You can also tell it's umask related as the RID and agent-info files are created with global read permissions 0644 which for something like OSSEC, I would say global perms should be avoided.

By the way, I hope it does not seem like I am criticizing this project, I absolutely love the idea!  OSSEC desperately needed to be brought into the 21st century and this is awesome, totally glad to have stumbled across it.

**Comment by vikman90** [comment]
Hi @bdwyertech,

do not worry about the rIDs permissions: they must have _write_ permissions for the owner _ossecr_ since `ossec-remoted` is the only process that should write on them.

The API (running with user _ossec_) needs only to erase those files: a file deletion is a _write_ operation on the container folder, so your fix should be enough to solve this issue.

Thank you very much for your PR and your feedback.

Best regards.
Victor.

**Comment by bdwyertech** [comment]
@vikman90 folder permissions do not dictate the permissions of newly created files, the umask of the process creating them does, so the daemon will continue writing them as 0644.

Elsewhere in the code the umask is explicitly declared.  Example: https://github.com/wazuh/wazuh/blob/1b6de44e9dc70fb056f804ebac58cabd9a357912/src/monitord/sign_log.c#L35

Basically the remoted daemon needs to have umask 0002  set somewhere in the function that creates those files.

**Comment by bdwyertech** [comment]
This better explains umask vs folder permissions

https://linuxaria.com/article/linux-shell-understanding-umask-with-examples

**Comment by vikman90** [comment]
@bdwyertech thank you for the explanation. Let's go back to the permissions list (after your fix):

```
drwxrwx--- 2 ossecr ossec queue/rids
-rw-r--r-- 1 ossecr ossec queue/rids/sender_counter
-rw-r--r-- 1 ossecr ossec queue/rids/001
-rw-r--r-- 1 ossecr ossec queue/rids/002
```

We do not care about the permissions for _others_ at files since the folder `rids` has no permissions for other users so the could not reach them.

However the folder `rids` has full permissions for the group. If we set the `umask` to `0002` files would be created with permissions:

```
-rw-rw-r-- 1 ossecr ossec queue/rids/sender_counter
-rw-rw-r-- 1 ossecr ossec queue/rids/001
-rw-rw-r-- 1 ossecr ossec queue/rids/002
```

This would allow users belonging to group `ossec` –such the API or other Wazuh processes– to modify these files. We do not desire this because it could lead to file corruption. No other process than `ossec-remoted` should write on those files. In other words, if we changed the `umask` we would have to do `chmod` on every new file in order to restrict _write_ permissions. 

Despite of this, users belonging to group `ossec` are able to remove those files since they have _write_ permissions on the folder `rids`. They cannot modify them but can delete them.
